script/build.js

```diff
@@ -690,6 +691,40 @@ smith.use(redirect({
   '/education/apply-for-education-benefits/': '/education/apply/'
 }));
 
+/*
+Add nonce attribute with substition string to all inline script tags
+Convert onclick event handles into nonced script tags
+*/
+smith.use((files, metalsmith, done) => {
+  Object.keys(files).forEach((file) => {
+    if (path.extname(file) !== '.html') return;
+
+    const data = files[file];
+    const dom = new jsdom.JSDOM(data.contents.toString());
+    dom.window.document.querySelectorAll('script').forEach((scriptEl) => {
+      if (scriptEl.textContent !== '') {
+        scriptEl.setAttribute('nonce', '**CPS_NONCE**');
+      }
+    });
+    dom.window.document.querySelectorAll('[onclick]').forEach((onclickEl) => {
+      if (onclickEl.id === '') {
+        onclickEl.id = Math.random().toString(36).replace(/[^a-z]+/g, '').substr(0, 10);
+      }
+      const id = onclickEl.id;
+      const onclick = onclickEl.getAttribute('onclick');
+      const newScript = dom.window.document.createElement('script');
+      newScript.setAttribute('nonce', '**CSP_NONCE**');
+      newScript.textContent = `(function() { var e = document.getElementById('${id}'); e.addEventListener('click', function(ev) { ${onclick} }); })();`;
+      onclickEl.removeAttribute('onclick');
+      onclickEl.appendChild(newScript);
+    });
```

Comment:
Are there a lot of instances of defining the onclick handler in the element like this? This is cool code but I think it would be better to define a standard for not defining onclick handlers in the element, because this seems a little magical/cryptic for a build step.

Reply:
I totally agree that this is probably too magic. There are just 97 instances of onclick handlers (doesn't seem like there are any other inline event handlers being used). I can manually remove them all and turn this into a step that will error if elements have inline handlers instead. However, I think its possibly fine to keep this build step for now and gradually remove the handlers along with making it a rule to not add any more.

Reply:
Yeah that sounds right. I think the navigation alone has a lot of instances for capturing analytics. But that makes sense to me. @jbalboni what do you think?

Reply:
It may be hard to get away from them entirely in the static content, but it seems like we could at least handle the analytics use case differently.